### PR TITLE
fix: Accept CTAS with function

### LINKED ISSUE #253: Unable to extract alias of columns using function with CTAS
While `INSERT OVERWRITE TABLE` can extract aliases for columns with function, `CREATE TABLE ... AS` can't.

Expected columns:

```sh
$ sqllineage -l column -e "INSERT OVERWRITE TABLE tbl1
SELECT
  coalesce(t1.col1, 0) AS col1,
  IF(
    t1.col1 IS NOT NULL,
    1,
    NULL
  ) AS col3
FROM
  t1
"
<default>.tbl1.col1 <- <default>.t1.col1
<default>.tbl1.col3 <- <default>.t1.col1
```

Actual:

```sh
$ sqllineage -l column -e "CREATE TABLE tbl1 AS
SELECT
  coalesce(t1.col1, 0) AS col1,
  IF(
    t1.col1 IS NOT NULL,
    1,
    NULL
  ) AS col3
FROM
  t1
"
<default>.tbl1.coalesce <- <default>.t1.coalesce
<default>.tbl1.if <- <default>.t1.if
```

### PR DESCRIPTION
**Pull request by chezou**
Monkey patch covering for https://github.com/andialbrecht/sqlparse/pull/662
close #253

### CODE REVIEW

File: sqllineage/utils/sqlparse.py [hunk 1]
```diff
@@ -128,6 +128,24 @@ def post(tlist, pidx, tidx, nidx):
     )
 
 
+@recurse(Function)
+def group_functions_as(tlist):
```
Comment (reata) [inline sqllineage/utils/sqlparse.py:132]: Can you add some comments to help understanding for this function?

I know I should do better for group_window function above, sorry.

**Review by reata — COMMENTED** [review]
Please add some comment and meanwhile allow me some time to understand how you want to tweak the AST.

File: sqllineage/utils/sqlparse.py (hunk 1 above)
Comment (chezou) [reply sqllineage/utils/sqlparse.py:132]: Added a comment  813598f
Please let me know if it suffices.

Comment (reata) [reply sqllineage/utils/sqlparse.py:132]: Yeah, the comment looks good.
I'll spend sometime to further understand this issue during weekend and then get back to you. Forgive me for being maybe over-cautious with monkey patching.

Comment (chezou) [reply sqllineage/utils/sqlparse.py:132]: Added more strict condition to scope the target [8db1b82](https://github.com/reata/sqllineage/pull/258/commits/8db1b82d23ce151f6d91bb4ec283b8c208eb47b4)

**Review by reata — APPROVED** [review]
LGTM

**Comment by chezou** [comment]
JFYI, This patch will be no more required in sqlparse 0.4.3 since my patch to upstream was merged.

**Comment by reata** [comment]
Thank you for the heads-up. I'll close watch the sqlparse 0.4.3 release.

**Pull request merged**